File: system-test/dotnet/Dsl.Core/UseCase/External/Clock/UseCases/ReturnsTime.cs
using Dsl.Core.External.Clock.UseCases.Base;
using Driver.Port.External.Clock;
using Driver.Port.External.Clock.Dtos;
using Common;
using Dsl.Core.Shared;

namespace Dsl.Core.External.Clock.UseCases;

public class ReturnsTime : BaseClockCommand<VoidValue, VoidVerification>
{
    private string? _time;

    public ReturnsTime(IClockDriver driver, UseCaseContext context)
        : base(driver, context)
    {
    }

    public ReturnsTime Time(string? time)
    {
        _time = time;
        return this;
    }

    public override async Task<ClockUseCaseResult<VoidValue, VoidVerification>> Execute()
    {
        var request = new ReturnsTimeRequest
        {
            Time = _time
        };

        var result = await _driver.ReturnsTimeAsync(request);

        return new ClockUseCaseResult<VoidValue, VoidVerification>(
            result,
            _context,
            (response, ctx) => new VoidVerification(response, ctx));
    }
}

File: system-test/java/src/main/java/com/optivem/shop/dsl/core/usecase/external/clock/usecases/ReturnsTime.java
package com.optivem.shop.dsl.core.usecase.external.clock.usecases;

import com.optivem.shop.dsl.driver.port.external.clock.ClockDriver;
import com.optivem.shop.dsl.driver.port.external.clock.dtos.ReturnsTimeRequest;
import com.optivem.shop.dsl.core.usecase.external.clock.usecases.base.BaseClockUseCase;
import com.optivem.shop.dsl.core.shared.UseCaseResult;
import com.optivem.shop.dsl.core.shared.UseCaseContext;
import com.optivem.shop.dsl.core.shared.VoidVerification;

public class ReturnsTime extends BaseClockUseCase<Void, VoidVerification> {
    private String time;

    public ReturnsTime(ClockDriver driver, UseCaseContext context) {
        super(driver, context);
    }

    public ReturnsTime time(String time) {
        this.time = time;
        return this;
    }

    @Override
    public UseCaseResult<Void, VoidVerification> execute() {
        var request = ReturnsTimeRequest.builder()
                .time(time)
                .build();

        var result = driver.returnsTime(request);

        return new UseCaseResult<>(result, context, VoidVerification::new);
    }
}
